python/ray/tune/experiment/trial.py

```diff
@@ -292,42 +292,23 @@ def __init__(
         # Trial config
         self.trainable_name = trainable_name
         self.trial_id = Trial.generate_id() if trial_id is None else trial_id
-        self.config = config or {}
         self._local_dir = local_dir  # This remains unexpanded for syncing.
 
+        self.config = config or {}
+        # Save a copy of the original unresolved config so that we can swap
+        # out and update any reference config values after restoration.
+        self.__unresolved_config = self.config
```

Comment:
OOC, why double underscore?

Reply:
very private. nobody should touch or have access to this variable outside of Trial. under the hood, python replace this variable with a name classname__variablename__.